core/include/vm.h

```diff
@@ -63,6 +63,7 @@ struct vm_t {
 #define VM_FEATURES_FASTMMIO_BASIC 0x1
 #define VM_FEATURES_FASTMMIO_EXTRA 0x2
     uint32_t features;
+    uint32_t qemu_api_version;
```

Comment:
Would you consider to set it like VM_FEATURES_FASTMMIO_BASIC and VM_FEATURES_FASTMMIO_EXTRA by just adding a feature bit in vm_t, and set it in hax_vm_set_qemuversion()? So it is controlled by feature bit instead of version directly. You can still keep the codes in QEMU part but not add the member qemu_api_version in vm_t.

Reply:
Version is checked only on cr8 access, add VM_FEATURES_CR8 flag?